src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/migration/Cas2NoteMigrationJob.kt

```diff
@@ -0,0 +1,50 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.migration
+
+import org.slf4j.LoggerFactory
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Slice
+import org.springframework.transaction.support.TransactionTemplate
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas2ApplicationNoteEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas2ApplicationNoteRepository
+
+class Cas2NoteMigrationJob(
+  private val noteRepository: Cas2ApplicationNoteRepository,
+  private val transactionTemplate: TransactionTemplate,
+  private val pageSize: Int,
+) : MigrationJob() {
+  private val log = LoggerFactory.getLogger(this::class.java)
+  override val shouldRunInTransaction = true
+
+  override fun process() {
+    log.info("Starting CAS2 note migration process...")
+
+    var hasNext = true
+    var slice: Slice<Cas2ApplicationNoteEntity>
+    var page = 1
+    var notesWithoutAssessment = noteRepository.countAllNotesWithoutAssessment()
+    val notesWithAssessment = noteRepository.countAllNotesWithAssessment()
+
+    while (hasNext) {
+      log.info("Getting page $page for max page size $pageSize")
+      slice = noteRepository.findAllNotesWithoutAssessment(PageRequest.of(0, pageSize))
+      slice.content.forEach { note ->
+        transactionTemplate.executeWithoutResult {
+          log.info("Adding assessment to note ${note.id}")
+          note.assessment = note.application.assessment
+          noteRepository.save(
+            note,
+          )
+        }
+      }
+      hasNext = slice.hasNext()
+      page += 1
+    }
+    log.info("Before the migration:")
+    log.info("notes without assessments: $notesWithoutAssessment")
```

Comment:
> Are we confident that log.info will work on jobs? I remember the cache warming jobs didn't work when running on a real environment, though it might have been fixed since @davidatkinsuk was asking about this recently.

I haven't changed anything in this area but i'm pretty sure info logging is only an issue when the thread wasn't spawned by an HTTP/API request. So in this case i think logging should work fine